de.metas.marketing/base/src/main/java/de/metas/marketing/base/model/ContactPersonRepository.java

```diff
@@ -195,4 +207,42 @@ private static ContactPerson asContactPerson(@NonNull final I_MKTG_ContactPerson
 				.contactPersonId(ContactPersonId.ofRepoId(contactPersonRecord.getMKTG_ContactPerson_ID()))
 				.build();
 	}
+
+	public void createUpdateConsent(
+			@NonNull final ContactPerson contactPerson)
+	{
+		final I_MKTG_Consent consentExistingRecord = getConsentRecord(contactPerson);
+
+		final I_MKTG_Consent consent = consentExistingRecord != null ? consentExistingRecord : newInstance(I_MKTG_Consent.class);
+
+		consent.setAD_User_ID(contactPerson.getAdUserId());
```

Comment:
if `contactPerson.getAdUserId()` returns zero, then this will end up as the "system" user. 
please call `consent.setAD_User(null)` if `contactPerson.getAdUserId()==0`

Reply:
Fixed